### ISSUE TITLE
MAP 0 on training from scratch on COCO dataset 

### ISSUE THREAD

**Issue by Dev2022** [eriklindernoren/PyTorch-YOLOv3#92]
Hi , I have trained on COCO dataset with epoch 30. After training I test the weight. But the map was 0. Can anyone help me?

**Comment by perrywu1989** [comment]
What are your params and the loss of the end?

**Comment by Dev2022** [comment]
Hi Used the same 

batch=16
subdivisions=1
width=416
height=416
channels=3
momentum=0.9
decay=0.0005
angle=0
saturation = 1.5
exposure = 1.5
hue=.1

learning_rate=0.001
burn_in=1000
max_batches = 500200
policy=steps
steps=400000,450000
scales=.1,.1

**Comment by Dev2022** [comment]
@perrywu1989  The loss was 2 at the end of 30 epoch

**Comment by perrywu1989** [comment]
do you train on the latest version? I found the computed loss had changed

**Comment by Dev2022** [comment]
@perrywu1989  I have used the latest version.

**Comment by perrywu1989** [comment]
@Dev2022 it is because the mAP metric has changed. If you use the  mAP metric of older version. The result is 0.58.

**Comment by Dev2022** [comment]
@perrywu1989 Ok ..But what about the detect.py file. Because can not detect bounding box with the new trained weights also.

**Comment by perrywu1989** [comment]
@Dev2022 I think there is something wrong in the loss function which leads to the bad results you got. I am working on it now.

**Comment by Dev2022** [comment]
@perrywu1989  Thanks a lot :)

**Comment by beckhamchen** [comment]
The same problem with you. Have you solved it?

**Comment by Dev2022** [comment]
@beckhamchen yes same issue. No have not solved yet.

**Comment by sporterman** [comment]
> @perrywu1989 I have used the latest version. But with the yolov3.weights (provided by trained) the map is is .409

hello, i've trained coco on this code without any change. after 17 epoches, the recall was 0.6, but the precision keep 10-3 from the begining and never rise, what about your training output?

**Comment by RickvanHek** [comment]
Have the same problem, but then on my own dataset. Anyone has solved it?

**Comment by dtmoodie** [comment]
Hello, just to chime in, I am also having this issue.
I have pretrained weights from coco that I've confirmed to detect currently with detect.py, I then loaded these weights for training to see what the output would be during training and I found the loss to be high and the precision to be low.
I think there is either something wrong with the data processing, parsing, or the loss function.

**Comment by esrasiler** [comment]
Hi, we have done the same thing with @dtmoodie , and we have high recall, high loss and low precision.
Did anyone solve the problem?

**Comment by fourth-archive** [comment]
I've successfully trained YOLOv3 with PyTorch from scratch using the repo below. It does automatic image augmentation also, and approaches the darknet mAP after about 65 epochs:
https://github.com/ultralytics/yolov3

**Comment by Xudangliatiger** [comment]
Have the same problem，has anyone successfully changed the code？

**Comment by RickvanHek** [comment]
> Have the same problem，has anyone successfully changed the code？

Try the version of ultralytics, currently basing my master's thesis on that repository. Many updates, the author is friendly, and it works great, only downside is that it doesn't support multi gpu

[https://github.com/ultralytics/yolov3](You can find the repository here)

**Comment by Xudangliatiger** [comment]
Thanks for telling me that. I was trying to train my own dataset, but the bugs keep in my way. I will try that version, thanks again!

发送自 Windows 10 版邮件<https://go.microsoft.com/fwlink/?LinkId=550986>应用

________________________________
发件人: Rick van Hek <notifications@github.com>
发送时间: Monday, March 4, 2019 9:24:45 PM
收件人: eriklindernoren/PyTorch-YOLOv3
抄送: XDL; Comment
主题: Re: [eriklindernoren/PyTorch-YOLOv3] MAP 0 on training from scratch on COCO dataset (#92)


Have the same problem，has anyone successfully changed the code？

Try the version of ultralytics, currently basing my master's thesis on that repository. Many updates, the author is friendly, and it works great, only downside is that it doesn't support multi gpu

[https://github.com/ultralytics/yolov3](You can find the repository here)

―

**Comment by eriklindernoren** [comment]
Hey. This should be resolved in the latest version. :)
